### PR TITLE
1828: Get the waterfall auction step explicitly. 

### PR DESCRIPTION
**Pull request by crericha**
It is not active in multiplayer games until the game starts.

### CODE REVIEW

File: lib/engine/game/g_1828.rb [hunk 1]
```diff
@@ -268,7 +268,7 @@ def remove_extra_private_companies
                              .take(7 - @players.size)
         to_remove.each do |company|
           company.close!
-          @round.active_step.companies.delete(company)
+          @round.steps.find { |step| step.is_a?(Step::G1828::WaterfallAuction) }.companies.delete(company)
```
Comment (tobymao) [inline lib/engine/game/g_1828.rb:271]: should this be out of the hot loop?

Comment (crericha) [reply lib/engine/game/g_1828.rb:271]: The loop is executed a maximum of four times, over two steps and only called during game setup, so I didn't optimize; however, if you want me to. I can cache the step. Let me know.

**Comment by tobymao** [comment]
if it only happens during the setup thats ok

File: lib/engine/game/g_1828.rb (hunk 1 above)
Comment (crericha) [reply lib/engine/game/g_1828.rb:271]: Yeah, I try to avoid optimizations that won't have an impact on performance.

**Pull request merged**